File: c#/PainterVector1/PainterVector/XCommand.cs
namespace PainterVector
{
	public class XCommand
	{
		public XData xData = new XData();

		public void SetXData(XData xData)
		{
			this.xData = xData;
		}

		public void SetColor()
		{
			Utilities.SetColor(ref xData.color);
		}

		public void SetLineWidth(int lineWidth)
		{
			xData.lineWidth = lineWidth;
		}

		public void SetFigureType(XData.FigureType figureType)
		{
			xData.type = figureType;
		}
	}
}

File: web/painter/js/XCommand.js
class XCommand {
	constructor(tabs) {
		this.pDraw = null;
	}

	get getArr() {
		return this.tabs;
	}

	setActivePDraw(pDraw) {
		this.pDraw = pDraw;
		console.log(pDraw);
	}

	setColor(color) {
		console.log(color);
	}

	setType(type) {
		this.pDraw.xData.setType(type);
		console.log(type);
	}
}